sdk/src/Core/Amazon.Runtime/Credentials/InstanceProfileAWSCredentials.cs

```diff
@@ -147,6 +148,90 @@ protected override CredentialsRefreshState GenerateNewCredentials()
             return state;
         }
 
+        protected override async Task<CredentialsRefreshState> GenerateNewCredentialsAsync()
+        {
+            CredentialsRefreshState newState = null;
+            var token = await EC2InstanceMetadata.FetchApiTokenAsync().ConfigureAwait(false);
+
+            try
+            {
+                // Attempt to get early credentials. OK to fail at this point.
+                newState = await GetRefreshStateAsync(token).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                HttpStatusCode? httpStatusCode = ExceptionUtils.DetermineHttpStatusCode(e);
+
+                if (httpStatusCode == HttpStatusCode.Unauthorized)
+                {
+                    Logger.GetLogger(typeof(EC2InstanceMetadata)).Error(e, "EC2 Metadata service returned unauthorized for token based secure data flow.");
+                    throw;
+                }
+
+                var logger = Logger.GetLogger(typeof(InstanceProfileAWSCredentials));
+                logger.InfoFormat("Error getting credentials from Instance Profile service: {0}", e);
+
+                // if we already have cached credentials, we'll continue to use those credentials,
+                // but try again to refresh them in 2 minutes.
+                if (null != _currentRefreshState)
+                {
+#pragma warning disable CS0612, CS0618 // Type or member is obsolete
+                    var newExpiryTime = AWSSDKUtils.CorrectedUtcNow + TimeSpan.FromMinutes(2);
```

Comment:
I know you got this from existing code, but `AWSSDKUtils.CorrectedUtcNow` is not deprecated anymore so the `pragma warning disable` are redundant. You should remove them.